src/function/algebra/simplify/resolve.js

```diff
@@ -29,38 +31,59 @@ export const createResolve = /* #__PURE__ */ factory(name, dependencies, ({
    *     math.simplify.resolve(math.parse('x+y'), {x:1, y:2}) // Node {1 + 2}
    *     math.simplify('x+y', {x:2, y:'x+x'}).toString()      // "6"
    *
+   * See also:
+   *
+   *     simplify, evaluate
+   *
    * @param {Node} node
    *     The expression tree to be simplified
-   * @param {Object} scope with variables to be resolved
+   * @param {Object} scope
+   *     Scope specifying variables to be resolved
+   * @return {Node} Returns `node` with variables recursively substituted.
+   * @throws {ReferenceError}
+   *     If there is a cyclic dependency among the variables in `scope`,
+   *     resolution is impossible and a ReferenceError is thrown.
    */
-  function resolve (node, scope) {
+  function resolve (node, scope, within = []) { // note `within` not documented
+    // (since `within` is for internal cycle detection only).
     if (!scope) {
       return node
     }
     if (!isMap(scope)) {
       scope = createMap(scope)
     }
     if (isSymbolNode(node)) {
+      if (within.indexOf(node.name) !== -1) {
```

Comment:
Generally I prefer not to mutate function input arguments, but here with `within` (and only internally) it makes sense.

Would it make sense for performance to change make `within` a `Set` instead of an `Array`?

Reply:
Ah I was not thinking of a case with 100s of variables, but I suppose someone may want to use mathjs for such a case. Will change.